hudi-cli/src/main/java/org/apache/hudi/cli/commands/CommitsCommand.java

```diff
@@ -232,18 +211,13 @@ public String showCommitFiles(@CliOption(key = {"commit"}, help = "Commit to sho
     return HoodiePrintHelper.print(header, new HashMap<>(), sortByField, descending, limit, headerOnly, rows);
   }
 
-  @CliAvailabilityIndicator({"commits compare"})
-  public boolean isCompareCommitsAvailable() {
-    return HoodieCLI.tableMetadata != null;
-  }
-
   @CliCommand(value = "commits compare", help = "Compare commits with another Hoodie dataset")
   public String compareCommits(@CliOption(key = {"path"}, help = "Path of the dataset to compare to") final String path)
       throws Exception {
 
     HoodieTableMetaClient target = new HoodieTableMetaClient(HoodieCLI.conf, path);
     HoodieTimeline targetTimeline = target.getActiveTimeline().getCommitsTimeline().filterCompletedInstants();
-    HoodieTableMetaClient source = HoodieCLI.tableMetadata;
+    HoodieTableMetaClient source = HoodieCLI.getTableMetaClient();
```

Comment:
can you help me understand when exactly isCompareCommitsAvailable() will be called? If it is going to be called at the beginning, shouldn't we move this line to line no 217?

Reply:
> can you help me understand when exactly isCompareCommitsAvailable() will be called? If it is going to be called at the beginning, shouldn't we move this line to line no 217?

Yes, isCompareCommitsAvailable() called at first. Need to move it to 217. I will fix it.

Reply:
> can you help me understand when exactly isCompareCommitsAvailable() will be called? If it is going to be called at the beginning, shouldn't we move this line to line no 217?

Yes, `isCompareCommitsAvailable()` was be called at first, had move this line to first line.